plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeGcsConnectorSmokeTest.java

```diff
@@ -0,0 +1,302 @@
+    }
+
+    @Override
+    protected List<String> listCheckpointFiles(String transactionLogDirectory)
+    {
+        return listAllFilesRecursive(transactionLogDirectory).stream()
+                .filter(path -> path.contains("checkpoint.parquet"))
+                .collect(toImmutableList());
+    }
+
+    private List<String> listAllFilesRecursive(String directory)
+    {
+        String path = bucketUrl() + directory;
+        ImmutableList.Builder<String> paths = ImmutableList.builder();
+
+        try {
+            RemoteIterator<LocatedFileStatus> files = fileSystem.listFiles(new org.apache.hadoop.fs.Path(path), true);
+            while (files.hasNext()) {
+                LocatedFileStatus file = files.next();
+                if (!file.isDirectory()) {
+                    paths.add(file.getPath().toString());
+                }
+            }
+            return paths.build();
+        }
+        catch (FileNotFoundException e) {
+            return ImmutableList.of();
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    protected String bucketUrl()
+    {
+        return format("gs://%s/%s/", gcpStorageBucket, bucketName);
+    }
+
+    // These overrides are required because the error message does not match the standard format
+    @Override
+    @Test
+    public void testInsert()
+    {
+        assertQueryFails("INSERT INTO region (regionkey) VALUES (42)", "Inserts are not supported on the gs filesystem");
+    }
+
+    @Override
+    @Test
+    public void testUpdate()
+    {
+        assertQueryFails("UPDATE nation SET nationkey = nationkey + regionkey WHERE regionkey < 1", "Updates are not supported on the gs filesystem");
+    }
+
+    @Override
+    @Test
+    public void testMerge()
+    {
+        assertQueryFails("MERGE INTO nation n USING nation s ON (n.nationkey = s.nationkey) " +
+                        "WHEN MATCHED AND n.regionkey < 1 THEN UPDATE SET nationkey = 5",
+                "Updates are not supported on the gs filesystem");
+    }
+
+    @Override
+    @Test
+    public void verifySupportsDeleteDeclaration()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_supports_delete", "AS SELECT * FROM region")) {
+            assertQueryFails("DELETE FROM " + table.getName(), "Deletes are not supported on the gs filesystem");
+        }
+    }
+
+    @Override
+    @Test
+    public void verifySupportsRowLevelDeleteDeclaration()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_supports_row_level_delete", "AS SELECT * FROM region")) {
+            assertQueryFails("DELETE FROM " + table.getName() + " WHERE regionkey = 2", "Deletes are not supported on the gs filesystem");
+        }
+    }
+
+    @Override
+    @Test
+    public void testCreatePartitionedTable()
+    {
+        String tableName = "test_create_partitioned_table_" + randomTableSuffix();
```

Comment:
> These overrides are required because the error message does not match the standard format

This test doesn't have assertion for error message. Could you leave a comment why this test needs override?